From c02540b0451041e5915668be16524bed2b109f43 Mon Sep 17 00:00:00 2001
Message-Id: <c02540b0451041e5915668be16524bed2b109f43.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Rafael Barbalho <rafael.barbalho@intel.com>
Date: Wed, 23 Jul 2014 14:17:22 +0100
Subject: [PATCH 205/312] REVERTME [VPG]: drm/i915/chv: Add workaround for
 MIPI lanes on A silicon

A0-A3 silicon has bad clock timing on the MIPI data lanes. This is not expected
to be fixed until B0 silicon, so in the meantime there is a workaround by
jittering the data lanes.

Change-Id: I157eff2c5099bdd1ea9390ec2d95bdc0cd71062f
Signed-off-by: Rafael Barbalho <rafael.barbalho@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index 3ae49e5..6b159a9 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -147,6 +147,8 @@ static void intel_dsi_enable(struct intel_encoder *encoder)
 		/* assert ip_tg_enable signal */
 		temp = I915_READ(MIPI_PORT_CTRL(pipe)) & ~LANE_CONFIGURATION_MASK;
 		temp = temp | intel_dsi->port_bits;
+		if (IS_CHERRYVIEW(dev))
+			temp |= 0xe82d0000;
 		I915_WRITE(MIPI_PORT_CTRL(pipe), temp | DPI_ENABLE);
 		POSTING_READ(MIPI_PORT_CTRL(pipe));
 	}
-- 
1.7.9.5

